### ISSUE TITLE
Upgraded from 2.4.3 -> 2.5.2 - One RKE cluster I can't manage System project

### ISSUE THREAD

**Issue by spirilis** [rancher/rancher#30032]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
Not sure this is reproducible because it's only happening on one RKE cluster of mine, others don't seem to have this problem.  After upgrading to 2.5.2 (and trying downgrading to 2.5.1; same results) I see my "tools-dev" RKE cluster's System project cannot be selected in the Cluster Manager UI dropdown menu; it's greyed out with "Cluster not ready" under the project name.

**Other details that may be helpful:**
I can view this cluster's dashboard and the Cluster, Nodes tabs etc.  When I go into Projects/Namespaces, click on the 3-dot menu next to the System project and then View in API, I see some curious state at the bottom:

```
"name": "System",
"namespaceDefaultResourceQuota": null,
"namespaceId": null,
"podSecurityPolicyTemplateId": "unrestricted",
"resourceQuota": null,
"state": "error",
"transitioning": "error",
"transitioningMessage": "",
"type": "project",
"uuid": "eca454fe-78bd-11e9-9974-0050569feaa5"
```

**Environment information**
2.5.2 installed HA

**Cluster information**
- RKE on virtual machines hosted on a Rancher instance that itself is hosted on RKE nodes on VMs
- v1.18.3 and v1.18.10 (upgrading to v1.18.10 using Rancher 2.5.2 did not seem to resolve the issue)

- Docker version (use `docker version`):
```
Client:
 Version:           18.09.4
 API version:       1.39
 Go version:        go1.10.8
 Git commit:        d14af54266
 Built:             Wed Mar 27 18:34:51 2019
 OS/Arch:           linux/amd64
 Experimental:      false
```

**Comment by rmweir** [comment]
Are you able to get a kubeconfig for this cluster or access it through kubectl in the UI?

**Comment by spirilis** [comment]
Yup:

```
> kubectl -n kube-system get deploy
NAME                 READY   UP-TO-DATE   AVAILABLE   AGE
coredns              2/2     2            2           454d
coredns-autoscaler   1/1     1            1           454d
metrics-server       1/1     1            1           544d
```

By all accounts there doesn't seem to be anything wrong with the k8s runtime, services are running.  Just an oddity with rancher cluster manager UI where I can't view the project (but can view other projects, including Default).

**Comment by rmweir** [comment]
Can you share `kubectl get pods -n cattle-system`?

**Comment by rmweir** [comment]
Are you using the main admin user?

**Comment by spirilis** [comment]
Yes

**Comment by spirilis** [comment]
FYI, tried something-

I created a new project SystemZ, and tried moving all the namespaces over to it.  All would move but one - "cattle-logging".  The moment I unchecked cattle-logging, the "Move" button appeared in blue.  I've since migrated those and then renamed System to SystemL, and SystemZ to System.

I can now manage the System project and it's the "SystemL" project - whose only member is cattle-logging - that shows greyed with "Cluster not ready".

**Comment by spirilis** [comment]
Now the next problem- Can't delete the SystemL project, Rancher acknowledges this as "the" System project somehow.  Can this be remedied with an API call or update to something in Rancher?

**Comment by wzrdtales** [comment]
This is not related to system. All projects can be hit. upgrade from last 2.4.x to 2.5.2

![image](https://user-images.githubusercontent.com/1786821/99659231-d98c2700-2a60-11eb-8e33-e564f8c53a7b.png)

![api#](https://user-images.githubusercontent.com/1786821/99659378-12c49700-2a61-11eb-9f2c-af3f88e0cf86.png)

The project is in an error state for some reason.

**Comment by wzrdtales** [comment]
@superseb Any fix for this in sight? It is really annoying to be only able to access it from the projects tab or direct url.

**Comment by superseb** [comment]
@wzrdtales Is the `GrafanaDeployed` the only error in the `View in API` view? No other errors on the Project or Cluster?

@spirilis What are the errors in `View in API` for the Project/Cluster for your setup?

**Comment by wzrdtales** [comment]
this was the project and yes that is the only error. the namespace inside of this project has this status:

![image](https://user-images.githubusercontent.com/1786821/102344035-9bb8ea80-3f9b-11eb-8024-b96e45bbda94.png)

No error for the cluster whatsoever

**Comment by superseb** [comment]
Can you both share the full output of `View in API` from the project and cluster? And if available, the log from Rancher when the cluster was updated.

**Comment by danielhass** [comment]
@superseb we are hitting the same issue coming from Rancher v2.4.8 to v2.5.5. A single project has the same issue as described above. As we saw some error in the API `monitoringStatus` we tried to enable and disable the project monitoring but without success.

Sadly we do not have the Rancher log during the cluster update available but we can provide you with the full project API view (project and cluster id redacted): https://gist.github.com/danielhass/13521de23269c3c37cb9eb898f6633c0

Another strange behavior we saw is that we could move ~80% of the namespace in the project to a different one for two namespaces in the project overview the "Move" button is greyed out if we select the namespace.

Please let me know how we can help with further debugging this issue!

**Comment by wzrdtales** [comment]
@superseb obviously. no. i wont share such details in a public issue. i am open for a remote session though let me know when this would be possible

**Comment by danielhass** [comment]
Just a follow up for other that might stumble across this. After verifying that the project actually is in a fine state and we were able to access everything via the direct project link (not the drop-down in the Rancher UI) we isolated our situation to a erroneous transition condition on the receptive project which caused the UI to show the project as _not ready_.

We were able to remove the erroneous condition on k8s project API object that resides on the Rancher cluster itself. Afterwards we were able to access to project again via the UI.

However as in our case not the _System_ project was hit by this issue and we were not able to find the root cause for the failed transition this is more of a workaround than a real fix to the underlying problem.

**Comment by wzrdtales** [comment]
can you elaborate more what did you delete? I still have projects that appear broken although they're accessible over other menus

**Comment by danielhass** [comment]
@wzrdtales - as I can see from the screenshots you shared above your `GrafanaDeployed` condition is in state `false` with reason `error`. This is the same we had with our `MonitoringEnabled` condition.

Those conditions correspond to [Kubernetes object `status`](https://kubernetes.io/docs/concepts/overview/working-with-objects/kubernetes-objects/#object-spec-and-status) of the respective Rancher `Project` object on the Kubernetes cluster that hosts Rancher itself.

In the respective namespace that corresponds to the managed cluster that has the error you should be able to find several objects of the type `management.cattle.io/v3/Project`. The name of the object corresponds to the project ID with you should be able to find via the _View in API_ screen (or simply via looking at the URL in the project view). This object has several `conditions` in the `status` section. One of them should be your erroneous `MonitoringEnabled` condition. E.g:

```
apiVersion: management.cattle.io/v3
kind: Project
[...]
status:
  conditions:
  - lastUpdateTime: "2021-03-30T07:10:16Z"
    status: "True"
    type: InitialRolesPopulated
[...]
```
You should be able to delete this condition and if it doesn't come back your project should be accessible again via the UI just fine.

Let me know if this worked for you too.
